opslib/osm/pod.py

```diff
@@ -294,6 +309,69 @@ def build(self):
         return container
 
 
+class PodRestartPolicy:
+    """
+    Class that expresses which fields of the pod spec should force a pod restart
+
+    At the moment, only secrets can force restart the pod. More parts of the pod
+    can be added to the PodRestartPolicy while we need them.
+    """
+
+    default_policy = {
+        "kubernetesResources": {
+            "secrets": False,
+        },
+    }
+
+    def add_secrets(self, secret_names: Set[str] = None) -> NoReturn:
+        """
+        Add secrets to the restart policy
+
+        :param: secret_names: Set of secrets that will cause a force restart of the pod.
+                              The set should include the name of the secrets.
+                              If no secret_names are specified, all the secret will force
+                              the restart of the pod
+        """
+        self.default_policy["kubernetesResources"]["secrets"] = (
+            secret_names if secret_names else True
+        )
+
+    def _find_resources_with_name(
+        self, resources: Dict, resources_names: Set[str]
+    ) -> List[Any]:
+        resources_found = []
+        for name in resources_names:
+            resource = find_in_list_with_key(resources, key="name", value=name)
+            if resource:
+                resources_found.append(resource)
+        return resources_found
+
+    def policy_hash(self, pod_spec: Dict) -> str:
+        """
+        Return a hash of the Pod Spec taking into account the current policy.
+        This means that for the hash, only the data included in the policy will be added
+
+        :param: pod_spec: Pod spec
+        """
+        spec = pod_spec.copy()
+        policy_spec = {"kubernetesResources": {}}
+
+        for resource_name, resource_names in self.default_policy[
```

Comment:
Confusing naming of variables

Reply:
resource_type, resource_names would be better, no?

Reply:
done